internal/helper_test.go

```diff
@@ -0,0 +1,43 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package internal
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestUnsafeByteSlice2String(t *testing.T) {
+	tests := [][]byte{
+		[]byte("hello world"),
+		[]byte(`{"some":"json", "simple": true}`),
+		[]byte(`multi
+		line
+		test`),
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(string(test), func(t *testing.T) {
+			str := UnsafeByteSlice2String(test)
+			if strings.Compare(str, string(test)) != 0 {
```

Comment:
```suggestion
			if str != string(test) {
```

Reply:
FWIW I don't think I have ever used `string.Compare`. Had to go read the docs 😄 .

To quote the docs:

> It is usually clearer and always faster to use the built-in string comparison operators ==, <, >, and so on.